dowhy/causal_refuters/non_parametric_sensitivity_analyzer.py

```diff
@@ -0,0 +1,172 @@
+import scipy
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.model_selection import KFold
+from econml.utilities import cross_product
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.linear_model import LinearRegression, Lasso
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.compose import ColumnTransformer, make_column_selector
+from dowhy.utils.util import get_numeric_features
+from dowhy.causal_refuters.partial_linear_sensitivity_analyzer import PartialLinearSensitivityAnalyzer
+from dowhy.causal_refuters.reisz import ReiszRegressor, ReiszRepresenter, generate_moment_function, get_alpha_estimator, get_generic_regressor, create_polynomial_function
+
+
+class NonParametricSensitivityAnalyzer(PartialLinearSensitivityAnalyzer):
+    """
+    Class to perform Non parametric Senitivity Analysis
+    :param theta_s: point estimate for the estimator
+    
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.theta_s = kwargs["theta_s"] if "theta_s" in kwargs else 0
+
+        self.moment_function = None    
+        self.alpha_s = None
+        self.m_alpha = None
+        self.g_s = None
+        self.m_g = None
+
+    def check_sensitivity(self, plot=True):
+        """
+        Function to perform sensitivity analysis. 
+
+        :param plot: plot = True generates a plot of lower confidence bound of the estimate for different variations of unobserved confounding.
+                     plot = False overrides the setting
+        The following formulas are used in the analysis.

```

Comment:
this description should come before the :param:. After that, param and :returns can be written